File: sdk/dotnet/NextPayClient.cs
using System.Net.Http;
using System.Text;
using System.Threading.Tasks;

namespace PayForge {
  public class PayForgeClient {
    private readonly HttpClient _http;

    public PayForgeClient(string apiKey, string baseUrl = "https://api.payforge.local/api/v1") {
      _http = new HttpClient();
      _http.BaseAddress = new System.Uri(baseUrl.TrimEnd('/'));
      _http.DefaultRequestHeaders.Add("x-api-key", apiKey);
    }

    public async Task<string> CreatePaymentAsync(string payloadJson) {
      var content = new StringContent(payloadJson, Encoding.UTF8, "application/json");
      var response = await _http.PostAsync("/payments", content);
      response.EnsureSuccessStatusCode();
      return await response.Content.ReadAsStringAsync();
    }
  }
}

File: sdk/php/NextPayClient.php
<?php

class PayForgeClient
{
    private string $apiKey;
    private string $baseUrl;

    public function __construct(string $apiKey, string $baseUrl = 'https://api.payforge.local/api/v1')
    {
        $this->apiKey = $apiKey;
        $this->baseUrl = rtrim($baseUrl, '/');
    }

    private function request(string $method, string $path, ?array $payload = null): array
    {
        $ch = curl_init($this->baseUrl . $path);
        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
        curl_setopt($ch, CURLOPT_CUSTOMREQUEST, $method);
        curl_setopt($ch, CURLOPT_HTTPHEADER, [
            'Content-Type: application/json',
            'x-api-key: ' . $this->apiKey,
        ]);
        if ($payload !== null) {
            curl_setopt($ch, CURLOPT_POSTFIELDS, json_encode($payload));
        }

        $response = curl_exec($ch);
        $status = curl_getinfo($ch, CURLINFO_HTTP_CODE);
        curl_close($ch);

        $data = $response ? json_decode($response, true) : [];
        if ($status >= 400) {
            throw new Exception($data['error'] ?? 'payforge_request_failed');
        }

        return $data;
    }
}
